homework_1/task3/task3.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+)
+
+type pokemonAPIResponse struct {
+	Name                   string
+	LocationAreaEncounters string `json:"location_area_encounters"`
+}
+
+type pokemonEncounters struct {
+	LocationArea struct {
+		Name string
+	} `json:"location_area"`
+}
+
+type pokemonStruct struct {
+	Name     string
+	Location []string
+}
+
+func parseProgramArguments(pokemon *string) {
+	flag.StringVar(pokemon, "pokemon", "", "desired pokemon's name or number")
+
+	flag.Parse()
+}
+
+func fetchHTTPResponse(url string) (*http.Response, error) {
+	httpClient := pester.New()
+	return httpClient.Get(url)
+}
+
+func getResponseContent(response *http.Response) ([]byte, error) {
+	return ioutil.ReadAll(response.Body)
+}
+
+func getDataFromURL(url string) ([]byte, error) {
+
+	httpResponse, err := fetchHTTPResponse(url)
+	if err != nil {
+		return nil, errors.WithMessage(err, "fetching data from pokemon API")
+	}
+
+	bodyContent, err := getResponseContent(httpResponse)
+	if err != nil {
+		return nil, errors.WithMessage(err, "reading body of pokemon API response")
+	}
+
+	return bodyContent, nil
+}
+
+func marshalPokemonStruct(pokemon string, locations []pokemonEncounters) pokemonStruct {
+	var resultPokemon pokemonStruct
+
+	resultPokemon.Name = pokemon
+	for _, locationArea := range locations {
+		resultPokemon.Location = append(resultPokemon.Location, locationArea.LocationArea.Name)
+	}
+
+	return resultPokemon
+}
+
+func printResults(pokemon pokemonStruct) {
+	jsonToPrint, err := json.Marshal(pokemon)
+	if err != nil {
+		log.Fatal(
+			errors.WithMessage(err, "marshaling the result structure"),
+		)
```

Comment:
Functions should generally return the error as a value and let the caller handle it. There are some exceptions but those are rare. 

The functions here should return the error and the `main` function should handle those (probably by calling `log.Fatal`)

Reply:
Oops, that was an oversight. I will fix it now.